homeassistant/components/ozw/cover.py

```diff
@@ -28,7 +32,11 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
     def async_add_cover(values):
         """Add Z-Wave Cover."""
 
-        async_add_entities([ZWaveCoverEntity(values)])
+        if values.primary.command_class == CommandClass.BARRIER_OPERATOR:
+            async_add_entities([ZwaveGarageDoorBarrier(values)])
+
+        else:
+            async_add_entities([ZWaveCoverEntity(values)])
```

Comment:
Create the entity in the if else block and move the call to `async_add_entities` out of the if else block.